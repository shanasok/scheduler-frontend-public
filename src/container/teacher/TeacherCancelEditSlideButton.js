import React from 'react';
import { bindActionCreators } from "redux";
import { connect } from "react-redux";
import { setEditSlideFalse } from '../../redux/actions/editSlideAction';
import { getCourses } from "../../utils/coursesFetcherGraphQL";
import { setCoursesJson } from "../../redux/actions/coursesAction";
import {getLessonContent} from "../../utils/lessonContentFetcherGraphQL";

export function resetData(setCoursesJson, setEditSlideFalse, lessonId, setCurrentLessonContent){
    getCourses(setCoursesJson, setEditSlideFalse);
    getLessonContent(lessonId, setCurrentLessonContent);
}

class TeacherCancelEditSlideButton extends React.Component {

    render() {
        return <button className='cancelEditSlideButton curvedButton' onClick={
                () => resetData(this.props.actions.setCoursesJson,
                    this.props.actions.setEditSlideFalse,
                    this.props.lesson.id,
                    this.props.actions.setCurrentLessonContent
                )

            }>
                Exit Edit Mode
            </button>
    }
}

const mapStateToProps = state => ({
    lesson: state.currentLesson,
    slideEditMode: state.slideEditMode,
});

function mapDispatchToProps(dispatch) {
    return {
        actions: bindActionCreators({
            setCoursesJson,
            setEditSlideFalse,
        }, dispatch)
    }
}

export default connect(mapStateToProps, mapDispatchToProps)(TeacherCancelEditSlideButton);