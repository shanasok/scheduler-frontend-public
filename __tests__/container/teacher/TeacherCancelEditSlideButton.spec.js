import React from 'react';
import TeacherCancelEditSlideButton from '../../../src/container/teacher/TeacherCancelEditSlideButton';
import renderer from 'react-test-renderer';
import configureMockStore from "redux-mock-store";
import {Provider as ReduxProvider} from "react-redux";
import {configure, mount} from "enzyme";
import Adapter from "enzyme-adapter-react-16";
import {MemoryRouter} from "react-router";

const mockStore = configureMockStore()
const mockState = {
    slideEditMode: true,
    currentLesson: {
        id: 1,
    }
}
let store = mockStore(mockState)

it('renders correctly when setEditSlide boolean is true', () => {

    const tree = renderer
        .create( <ReduxProvider store={store}><TeacherCancelEditSlideButton/></ReduxProvider>)
        .toJSON();
    expect(tree).toMatchSnapshot();
});

//This test doesn't work
// it('resetData_calls_functions_passed_in_as_parameters',  () => {
//     const mockFunctionSetCoursesJson = jest.fn(setCoursesJson);
//     const mockFunctionSetEditSlideFalse = jest.fn(setEditSlideFalse);
//     const mockFunctionSetCurrentLessonContent = jest.fn(setCurrentLessonContent);
//
//     resetData(mockFunctionSetCoursesJson,
//         mockFunctionSetEditSlideFalse,
//         1,
//         mockFunctionSetCurrentLessonContent);
//
//     expect(mockFunctionSetEditSlideFalse).toHaveBeenCalledTimes(1);
//     expect(mockFunctionSetCoursesJson).toHaveBeenCalledTimes(1);
//     expect(mockFunctionSetCurrentLessonContent).toHaveBeenCalledTimes(1);
// });


configure({adapter: new Adapter()});
it('onClick of button it should call function one time', () => {
    let wrapper = mount(
        <MemoryRouter initialEntries={[ '/' ]}>
            <ReduxProvider store={store}>
                <TeacherCancelEditSlideButton/>
            </ReduxProvider>
        </MemoryRouter>
    );
    wrapper.find('.cancelEditSlideButton').simulate('click');

    expect(wrapper.find(TeacherCancelEditSlideButton)).toHaveLength(1);
    wrapper.unmount();
});