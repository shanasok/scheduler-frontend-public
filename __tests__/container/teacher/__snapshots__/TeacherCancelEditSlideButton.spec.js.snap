// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly when setEditSlide boolean is true 1`] = `
<div>
  <button
    className="cancelEditSlideButton curvedButton"
    onClick={[Function]}
  >
    Cancel Edits
  </button>
</div>
`;
